src/algorithms/optimization/peps_optimization.jl

```diff
@@ -144,13 +152,83 @@ function fixedpoint(
     return peps_final, env_final, cost, info
 end
 
-# Update PEPS unit cell in non-mutating way
-# Note: Both x and η are InfinitePEPS during optimization
+"""
+    peps_normalize(A::InfinitePEPS)
+
+Normalize the individual tensors in the unit cell of an `InfinitePEPS` such that they each
+have unit Euclidean norm.
+"""
+function peps_normalize(A::InfinitePEPS)
+    normalized_tensors = map(unitcell(A)) do t
+        return t / norm(t)
+    end
+    return InfinitePEPS(normalized_tensors)
+end
+
+"""
+    peps_retract(x, η, α)
+
+Performs a norm-preserving retraction of an infinite PEPS `A = x[1]` along `η` with step
+size `α`, giving a new PEPS `A´`,
+```math
+A' \\leftarrow \\cos(α ||η||) A + \\sin(α ||η||) \\frac{η}{||η||},
+```
+and corresponding directional derivative `ξ`,
+```math
+ξ = \\cos(α ||η||) η - \\sin(α ||η||) ||η|| A,
+```
+such that ``\\langle A', ξ \\rangle = 0``. If the initial state was properly normalized, we
+have that
+```math
+||A'|| = ||A|| = 1.
+```
+"""
```

Comment:
This assumes that `||A|| == 1`. If this were not the case, and you just want to preserve `||A||` no matter its value, you should have

```math
A' \\leftarrow \\cos(α ||η||/||A||) A + \\sin(α ||η||/||A||) \\frac{η * ||A||}{||η||},
```

i.e. you want the two vectors in that equation to have the same norm (and be orthogonal), such that they span define a circle.

Reply:
I normalized `A` at the very start of the optimization so it was safe to assume $||A|| = 1$ in that context, but I agree we probably want the retraction to be correct when this is not satisfied so it can be used in other contexts. I'll update it accordingly.

Reply:
Should be fixed now.